doc/source/getting-involved.rst

```diff
@@ -219,6 +219,37 @@ check recent tests known to be flaky.
 
 .. _`CI`: https://github.com/ray-project/ray/tree/master/ci/travis
 
+API stability
+-------------
+
+Ray provides stability guarantees for its public APIs in Ray core and libraries. The level of stability provided depends on how the API is annotated.
+
+.. autofunction:: ray.util.annotations.PublicAPI
+.. autofunction:: ray.util.annotations.DeveloperAPI
+.. autofunction:: ray.util.annotations.Deprecated
```

Comment:
Nice! Looks pretty good in the generated docs.

Reply:
Maybe in the "Deprecated" section we can talk about what the expectation would be for deprecation?

For example, can you expect the deprecation to happen over the course of 1 release, 2 releases? Is it always expected to be the case that the annotation will be applied before removal? etc